util/plugins/eu.esdihumboldt.util/src/eu/esdihumboldt/util/geometry/interpolation/ArcInterpolation.java

```diff
@@ -0,0 +1,253 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.geom.LineString;
+
+import de.fhg.igd.slf4jplus.ALogger;
+import de.fhg.igd.slf4jplus.ALoggerFactory;
+
+/**
+ * Interpolation of a geometry. Specifically for Arc
+ * 
+ * @author Arun
+ */
+public class ArcInterpolation extends Interpolation<LineString> {
+
+	private static final ALogger log = ALoggerFactory.getLogger(ArcInterpolation.class);
+
+	/**
+	 * Constructor
+	 * 
+	 * @param coordinates raw geometry coordinates
+	 * @param maxPositionalError maximum positional error
+	 */
+	public ArcInterpolation(Coordinate[] coordinates, double maxPositionalError) {
+		super(coordinates, maxPositionalError);
+	}
+
+	/**
+	 * @see eu.esdihumboldt.util.geometry.interpolation.Interpolation#validateRawCoordinates()
+	 */
+	@Override
+	protected boolean validateRawCoordinates() {
+		if (rawGeometryCoordinates.length != 3) {
+			log.error("Invalid arc geometry. Arc must be represented by 3 points in GML.");
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * @see eu.esdihumboldt.util.geometry.interpolation.Interpolation#interpolatedGeometry()
+	 */
+	@Override
+	protected LineString interpolatedGeometry() {
+		// Calculate center of Arc
+		Coordinate centerOfArc = calculateCenterPoint(rawGeometryCoordinates);
+
+		// Calculate radius of Arc
+		double radius = Math.sqrt(Math.pow((rawGeometryCoordinates[0].x - centerOfArc.x), 2)
+				+ Math.pow((rawGeometryCoordinates[0].y - centerOfArc.y), 2));
+
+		// return Line String Geometry
+		return interpolateToLineString(rawGeometryCoordinates, centerOfArc, radius);
+	}
+
+	private Coordinate calculateCenterPoint(Coordinate[] arcCoordinates) {
+
+		double aSlope = (arcCoordinates[1].y - arcCoordinates[0].y)
+				/ (arcCoordinates[1].x - arcCoordinates[0].x);
+
+		double bSlope = (arcCoordinates[2].y - arcCoordinates[1].y)
+				/ (arcCoordinates[2].x - arcCoordinates[1].x);
+
+		// we can get two lines using above slope let's say p1p2: (ya =
+		// m_a(x1-x0) + y0) and p2p3: (yb = m_a(x2-x1) + y1)
+
+		Coordinate AB_Mid = new Coordinate((arcCoordinates[0].x + arcCoordinates[1].x) / 2,
+				(arcCoordinates[0].y + arcCoordinates[1].y) / 2);
+		Coordinate BC_Mid = new Coordinate((arcCoordinates[1].x + arcCoordinates[2].x) / 2,
+				(arcCoordinates[1].y + arcCoordinates[2].y) / 2);
+
+		// The center of the circle is the intersection of the two lines
+		// perpendicular to and passing through the midpoints of the lines p1p2
+		// and p2p3.
+
+		double centerX = ((aSlope * bSlope * (BC_Mid.y - AB_Mid.y)) + (aSlope * BC_Mid.x)
+				- (bSlope * AB_Mid.x)) / (aSlope - bSlope);
+
+		double centerY = AB_Mid.y - ((centerX - AB_Mid.x) / aSlope);
+
+		return new Coordinate(centerX, centerY);
+	}
+
+	private LineString interpolateToLineString(Coordinate[] arcCoordinates, Coordinate center,
```

Comment:
Would this also work directly with an ArcByCenterPoint representation? Then it might sense to expose that method or a similar method to easier support that kind of representation.

I see that this method at least relies on three points, but maybe it can be made to work also with two points (or a third point can be calculated).